src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1101,7 +1134,7 @@ private unsafe static int ReadFileNative(IntPtr hFile, byte[] bytes, int offset,
                 return errorCode;
             }
 
-            private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset, int count)
+            private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset, int count, bool useFileAPIs)
```

Comment:
We will use FileAPIs in case we are in not in Unicode mode or if we are talking to a pipe\file. 

https://msdn.microsoft.com/en-us/library/windows/desktop/ms683457(v=vs.85).aspx talks about the limitation of each API.

> ReadConsole and WriteConsole can only be used with console handles; ReadFile and WriteFile can be used with other handles (such as files or pipes). ReadConsole and WriteConsole fail if used with a standard handle that has been redirected and is no longer a console handle.

> ReadConsole and WriteConsole behave exactly like ReadFile and WriteFile except that they can be used either as wide-character functions (in which text arguments must use Unicode) or as ANSI functions (in which text arguments must use characters from the Windows character set). Applications that need to maintain a single set of sources to support either Unicode or the ANSI character set should use ReadConsole and WriteConsole.